internal/cachevalue/cache.go

```diff
@@ -95,25 +121,20 @@ func (t *Cache[_]) ttl() time.Duration {
 	return ttl
 }
 
-func (t *Cache[I]) get(ttl time.Duration) (item I, ok bool) {
-	t.mu.RLock()
-	defer t.mu.RUnlock()
-	if t.valueSet {
-		item = t.value
-		if ttl <= 0 {
-			return item, true
-		}
-		if time.Since(t.lastUpdate) < ttl {
-			return item, true
+func (t *Cache[T]) update() {
+	val, err := t.Update()
```

Comment:
> Since this is locked can we check if the Update() is actually?

Something missing here.

> With NoWait implementation we have the risk of many updates blocked and happen serially instead we can unblock on first success right away and not have to spend time updating if the last update is already updated.

No. There will only ever be one update running. All others will unblock as well when the new result is ready.